### ISSUE TITLE
Extend blockquote alerts to support Obsidian AlertTitle

### ISSUE THREAD

**Issue by benbarbour** [gohugoio/hugo#12805]
In addition to custom alert typo (see #12801) it would also be nice to support [Obsidian's callout titles](https://help.obsidian.md/Editing+and+formatting/Callouts#Change+the+title) too. At least enough for me to use a blockquote render hook to support it. As is, if this markdown is sent to the render hook:

```markdown
> [!NOTE] Alerts are awesome
> Alerts, as you may have noticed, are an awesome way to callout information to the reader.
```

The the text "Alerts are awesome" (the "title") is not passed to the render hook at all; it just vanishes. It would be great if the render hook got `.AlertTitle` or something so I could implement an alert with a title myself, even if the default was to do nothing with it (current behavior).

However, the following illustrates how GitHub handles things.

----

> [!NOTE] Title Test
> Title Test 1: I have a title

> [!NOTE]
> Title Test 2: I have no title

----

As you can see, GitHub doesn't even render the blockquote as an alert if there's a title. So maybe Hugo's current behavior should be considered a bug (feel free to change the tag to "bug" if you agree). If a title is present then Hugo shouldn't consider the blockquote to be an alert at all; `.AlertType` should be empty and `.Text` should contain the entire contents of the blockquote. Fixing this would allow me to parse the title myself if I want to.

**Comment by bep** [comment]
I guess this relates to #12801 ... This feature was added to support "GitHub syntax" which is pretty wide spread. My question for both of these issues are: How big/widespread is Obsidian. I just heard about it days ago, so I'm guessing it's a pretty new product.

**Comment by benbarbour** [comment]
> how big / wide spread is Obsidian?

The first release was in March 2020 so it's not brand new. Other than that I can only answer with anecdotes that sum to "it seems quite popular".

However, I think that this should be split into two issues.

1. A bug with Hugo's implementation of GitHub alerts. Hugo acts differently than GitHub when there's any text after the square brackets on the first line. GitHub doesn't render an alert (just a normal blockquote), but Hugo does. Worse, Hugo swallows the extra text and it's just gone from both the rendered site and the render hook.
2. A proposal to add optional support for `{.AlertTitle}` like Obsidian, but I'd understand completely if supporting Obsidian features is not a goal of this project.

**Comment by bep** [comment]
>If you want to make this issue the bug I can actually follow the contribution guidelines and start with a feature request on the forum to make a proposal.

This needs to wait on #12801 - if that gets implemented, then this isn't a bug anymore. I'm not spending any time on a bug now if it becomes a feature tomorrow.

**Comment by benbarbour** [comment]
I can see doing the work at the same time, but it's a different issue, isn't it? This bug affects `NOTE`, `TIP`, and all the other valid Alert types while #12801 is about adding custom note types.

**Comment by bep** [comment]
Testing this, i see these variants:

```
> [!tip] Callouts can have custom titles
> Like this one.

> [!tip] Title-only callout

> [!faq]- Are callouts foldable?
> Yes! In a foldable callout, the contents are hidden when the callout is collapsed


> [!faq]+ Are callouts foldable?
> Yes! In a foldable callout, the contents are hidden when the callout is collapsed


> [!question] Can callouts be nested?
> > [!todo] Yes!, they can.
> > > [!example]  You can even use multiple layers of nesting.
```

So, I suggest we add:

* AlertTitle
* AlertSign (+,- or empty)

**Comment by bep** [comment]
Can you take a look at #12815 and see if that works for you?
